### DISCUSSION TITLE
User rights management

### DISCUSSION THREAD

**Discussion by K1ngLear** [mealie-recipes/mealie#843] [Feature Request]
Hi,

first I've to say thank you for this wonderful piece of software.

I've one suggestion and I hope you would consider it. It would be nice if there would be a user right management. So that only  the creator and an admin can change a recipe while comments should be open for all. So I could give my child access while they can't delete on of the stored recipes.

Thx for considering.

**Comment by hay-kot** [comment]
Please follow the feature template

https://github.com/hay-kot/mealie/issues/317

**Comment by K1ngLear** [comment]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Plese Describe The Problem To Be Solved**
_REPLACE-THIS-TEXT: Any  loged in user can change or delete recipes from any other user. If you want to give access, e.g. to your family, that they can comment or rate an recipe they get automatic full access to the recipes. So it's possible that they delete or alter a recipe by accident. 

**(Optional): Suggest A Solution**
I'am no programmer at all, so I've no idea how to realise the solution. In my opinion should it only possible to the creator or an admin to alter or delete an existing recipe. Comments and ratings meanwhile should be open to every user. 

If there are multiple solutions, please present each one separately. Save comparisons for the very end.)

**Additional Information**
Sorry, I've no idea what PR means.
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by hay-kot** [comment]
I've added a task for v1 for this 

- https://github.com/hay-kot/mealie/issues/862

**Reply by K1ngLear** [reply]
Thank you very much

**Comment by yousefmansy1** [comment]
I want to start off to say this project is super cool and useful.

As an expansion to this I think assigning individual read/write access to recipes based on user/group would be a great addition.

I saw the task for this feature was already merged. Not sure if this was already implemented along with this or if I should make a new feature request. 

Excellent work thanks!

**Reply by hay-kot** [reply]
There is a way for the recipe owner to lock a recipe to keep other from editing, that may get at what you're trying to accomplish. There isn't a more fine grained control of user access. If you have another idea you're welcome to create another feature request.

**Reply by yousefmansy1** [reply]
Is the editing lock something coming in v1.0.0?

**Reply by hay-kot** [reply]
Yep, already on the beta 

![CleanShot 2022-03-16 at 21 43 42](https://user-images.githubusercontent.com/64056131/158744822-4f9ed35b-136f-4b5a-b71e-797393fdae24.png)

Will allow only the owner (creator) to edit or unlock the recipe for the group.

**Reply by yousefmansy1** [reply]
Nice looks cool!
What's the release timeline for v1.0.0?
There's a bunch of cool stuff I'm looking forward to.

**Reply by hay-kot** [reply]
Soooooooonnnn. Hopefully.

I've made all the major database changes I need to so now moving forward we'll be using migrations so you won't have to import/export. There's still a lot of rough edges but it's usable. Sometime in the next week or so I'm planning to do a write-up and publish an official beta release (v1.0.0b1) and an open beta testing to collect some user feedback and continue to publish beta versions until I'm confident there aren't any game stopping bugs. 

For what it's worth, I've moved over completely to the nightly builds at home and it's totally usable for me.

**Reply by yousefmansy1** [reply]
Sounds great I'll be looking forward to it!

I just started using mealie on unraid and so far liking it.

I've got a few ideas I'd be willing to try to contribute.
I'm new to contributing to open-source (much less a tech stack I've never used) so we'll see 😅.
